test/e2e/main_test.go

```diff
@@ -8,7 +8,12 @@ import (
 func TestMain(m *testing.M) {
 	cleanNativeProcessorCache()
 
-	exitCode := m.Run()
+	n := newInProcessE2ENetwork()
+
+	m.Run()
+	exitCode := m.Run() // run twice so that any test assuming a clean slate will fail; e2es shouldn't assume anything about system state
```

Comment:
It's cool we run twice, although it's the slowest test and now we run it twice.. are we sure we always want to do this?